### PR TITLE
[LD-4717] Improve dynamodb autoscaling by using an open source project

### PR DESCRIPTION
**Pull request by faizshukri**

### CODE REVIEW

File: core/AutoScaleService.js [hunk 1]
```diff
@@ -145,28 +142,27 @@ class AutoScaleService {
...
       // loop through consumed capacity metrics
       metrics.forEach(point => {
-        this.logger.log('Received %s metric for %s:%s: Sum: %d Per second: %d',
-          point.metricName,
-          point.tableName,
-          point.indexName || 'No index',
-          point.Sum || 0,
-          (point.Sum || 0) / analyzeDuration.asSeconds()
-        );
+        // this.logger.log('Received %s metric for %s:%s: Sum: %d Per second: %d',
```
Comment (faizshukri) [inline core/AutoScaleService.js]: For reference

File: serverless.yml [hunk 2]
```diff
@@ -70,37 +81,37 @@ functions:
     events:
       - schedule: rate(${self:custom.rates.down})
 
-  configLoad:
```
Comment (faizshukri) [inline serverless.yml]: For reference

File: core/AutoScaleConfigRepository.js [hunk 3]
```diff
@@ -67,6 +67,16 @@ class AutoScaleConfigRepository {
     if (config.constructor.name === 'Object')
       config = AutoScaleConfig.create(config);
 
+    delete config.upDuration;
+    delete config.downDuration;
+
+    for(const key in config.indexes) {
```
Comment (ykt) [inline core/AutoScaleConfigRepository.js]: can we use array filter() here?

File: serverless.yml [hunk 4]
```diff
@@ -13,30 +14,31 @@ provider:
...
 
   configTableName: ddb-auto-scale-config
+  tableNameRegex: ^ld-production-
 
   iamRoleStatements:
     - Effect: Allow
       Action:
         - dynamodb:*
-      Resource: arn:aws:dynamodb:${self:custom.region}:*:table/*
+      Resource: "*"
```
Comment (ykt) [inline serverless.yml]: Why do we need to have wildcard here?

File: serverless.yml [hunk 5]
```diff
@@ -13,30 +14,31 @@ provider:
   runtime: nodejs6.10
 
   stage: dev
-  region: us-east-1
+  region: us-west-2
```
Comment (ykt) [inline serverless.yml]: should we parameterize this too, e.g.: as an environment variable?

File: core/AutoScaleConfigRepository.js (hunk 3 above)
Comment (faizshukri) [inline core/AutoScaleConfigRepository.js]: can, if `config.indexes` is array.  unfortunately, it was object.

File: serverless.yml (hunk 5 above)
Comment (faizshukri) [inline serverless.yml]: it is parameterized. But not as environment variable, instead as cli argument

eg: `sls deploy --stage production --region us-west-2 `

File: serverless.yml (hunk 4 above)
Comment (faizshukri) [inline serverless.yml]: Somehow the previous one don't have enough permission to run `listTables` command.

File: core/AutoScaleConfigRepository.js (hunk 3 above)
Comment (ykt) [inline core/AutoScaleConfigRepository.js]: Then if that the case we can use reduce function, something like
```
config.indexes = Object.keys(config.indexes).reduce((default, value, key) => {
            if (key !== 'upDuration' && key !== 'downDuration') {
                  default[key] = value
            }
}, {});
```

File: serverless.yml (hunk 5 above)
Comment (ykt) [inline serverless.yml]: got it

File: core/AutoScaleConfigRepository.js (hunk 3 above)
Comment (LongYC) [inline core/AutoScaleConfigRepository.js]: Trivial, a space between `for` and `(`.

File: core/AutoScaleConfigRepository.js [hunk 6]
```diff
@@ -67,6 +67,16 @@ class AutoScaleConfigRepository {
...
+      if(config.indexes.hasOwnProperty(key)) {
```
Comment (LongYC) [inline core/AutoScaleConfigRepository.js]: Trivial, space between `if` and `(`.

**Pull request merged**